lib/cretonne/src/ir/dfg.rs

```diff
@@ -657,6 +665,28 @@ impl DataFlowGraph {
                         })
     }
 
+    /// Removes the argument `val` from `ebb`. Returns true if the removal is sucessful.
+    pub fn remove_ebb_arg(&mut self, ebb: Ebb, val: Value) -> bool {
```

Comment:
I think this is worth its own PR. A couple points:

- Please include some unit tests in the `dfg::tests` module.
- You may want to take a look at an existing method like `replace_ebb_arg()` for some ideas. 
- You'll need to update the value table. Each EBB argument value knows its argument number, so you can't just slide arguments around. The `value_is_attached()` method should be useful for writing tests of this.
- It could be worthwhile to use semantics similar to the `Vec::swap_remove` method. It is easy to introduce accidental quadratic behavior without it. Of course, it needs to be clear how EBB arguments are moved around by this function so you can manipulate the predecessors accordingly.